quark/subproject.py

```diff
@@ -811,16 +816,46 @@ def url_from_directory(self, directory, include_commit=True):
     def _gitlab_setup(self, url):
         self.gitlab_url = "https://" + url.netloc  # Enforce HTTPS for now.
 
-        # Try private token variables, in order of importance. GITLAB_PRIVATE_TOKEN is commonly used
-        # by python-gitlab's "gitlab" CLI tool, whereas GITLAB_TOKEN is used by GitLab's "glab" CLI
-        # tool.
-        env_vars = ("QUARK_GITLAB_PRIVATE_TOKEN", "GITLAB_PRIVATE_TOKEN", "GITLAB_TOKEN")
+        # Retrieve the token from the system's keyring if the "keyring" package is
```

Comment:
I think that it's better to check the env variables first, and _then_ the keyring, so that it's possible to easily override the keyring locally if necessary.